File: internal/model/lease.go
package model

import "time"

type Lease struct {
	ID         string    `json:"id"`
	PropertyID string    `json:"property_id"`
	TenantID   string    `json:"tenant_id"`
	StartDate  time.Time `json:"start_date"`
	EndDate    time.Time `json:"end_date"`
	RentAmount float64   `json:"rent_amount"`
	Deposit    float64   `json:"deposit"`
	Status     string    `json:"status"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
}

File: web/src/tests/msw/factories/lease.ts
import type { TLease } from '@/domains/leases';

export const createMockLease = (overrides: Partial<TLease> = {}): TLease => ({
  id: 'l1',
  property_id: 'p1',
  tenant_id: 't1',
  start_date: '2026-01-01T00:00:00Z',
  end_date: '2027-01-01T00:00:00Z',
  rent_amount: 1500,
  deposit: 1500,
  status: 'active',
  created_at: '2026-01-01T00:00:00Z',
  updated_at: '2026-01-01T00:00:00Z',
  ...overrides,
});
